src/Features/Core/Portable/Workspace/BackgroundParser.cs

```diff
@@ -79,7 +85,9 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs args)
                     // this consumed around 2%-3% of the trace after some other optimizations I did. Most of that
                     // was actually walking the documents list since this was causing all the Documents to be realized.
                     // Since this is on the UI thread, it's best just to not do the work if we don't need it.
-                    if (oldProject.SupportsCompilation && !object.Equals(oldProject.ParseOptions, newProject.ParseOptions))
+                    if (SolutionCrawlerOptions.GetBackgroundAnalysisScope(newProject) != BackgroundAnalysisScope.ActiveFile &&
+                        oldProject.SupportsCompilation &&
+                        !object.Equals(oldProject.ParseOptions, newProject.ParseOptions))
                     {
                         foreach (var doc in newProject.Documents)
```

Comment:
thi needs commenting as well.  say the socpe *is* ActiveFile.  are we now going to miss a project change that would cause us to want to reanalyze that file?

Reply:
Ah thanks again for pointing this out. This was no doing the right thing, and we already handle skipping non-active files in active file only scope further down, so I was able to just revert this.